File: 1 - Two Sum/cpp/one-pass-hash/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int,int> seen;
        for(int i = 0; i < nums.size(); i++) {
            if(seen.find(target - nums[i]) != seen.end()) {
                ans.push_back(i);
                ans.push_back(seen[target - nums[i]]);
                return ans;
            }
            else {
                seen.insert({nums[i],i});
            }
        }
        return ans;
    }
};

File: 1 - Two Sum/one-pass-hash/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        ans = []
        for i in range(len(nums)):
            if (target - nums[i]) in seen.keys():
                ans.append(i)
                ans.append(seen[target-nums[i]])
            else:
                seen[nums[i]] = i

        return ans
